source/speech/speech.py

```diff
@@ -2784,9 +2800,27 @@ def getTableInfoSpeech(
 	textList=[]
 	if newTable:
 		columnCount=tableInfo.get("column-count",0)
+		# Translators: Sub-part of the compound string to report a table
 		rowCount=tableInfo.get("row-count",0)
-		# Translators: reports number of columns and rows in a table (example output: table with 3 columns and 5 rows).
-		text=_("table with {columnCount} columns and {rowCount} rows").format(columnCount=columnCount,rowCount=rowCount)
+		columnCountText = ngettext(
+			# Translators: Sub-part of the compound string to report a table
+			"{columnCount} column",
+			"{columnCount} columns",
+			columnCount,
+		).format(columnCount=columnCount)
+		rowCountText = ngettext(

```

Comment:
Perhaps all these repeated components should be factored out into helper functions that just take the count?

e.g. `_columnCountStr(count), _rowCountStr(count)`

Reply:
Sorry, I do not understand which part of the code I can put in a common helper function. Could you clarify please?

Reply:
The following component is repeated multiple times in this code, same with the row equivalent:

```
ngettext(
	"{columnCount} column",
	"{columnCount} columns",
	columnCount,
).format(columnCount=columnCount)
```

Reply:
@seanbudd, I do not see the point in making a helper function to just call `ngettext` and `format``; IMO it makes code less easy to read.
Instead, I have made a helper function to factor out the whole computation of table size announcement. See commit 70d32e48bc780e82b5630bf66c9c8a62b90c6304.

If you do not agree with this approach, I can revert this change.

Reply:
OK I got it. Done in 6db8c7d90d3f0f1ebc14929c2e13be4cea1c08b0.

Note that I have kept the function `_rowAndColumnCountText` that I had extracted from `getPropertiesSpeech` since it allows to reduce a bit its complexity. `getPropertiesSpeech` remains much too complex though.